main.go

```diff
@@ -1,156 +1,201 @@
 package main
 
 import (
-	"crypto/rand"
-	"crypto/rsa"
-	"crypto/x509"
-	"encoding/pem"
-	"fmt"
-	"os"
-
-	"github.com/Xfers/cloudhsm-service-go/crypto"
-)
-
-func main() {
-
-	// Set test keys and data
-	privatekey_pem, publickey_pem := generateTestKeys()
-	data := "test data that is to be signed"
-
-	// Digest
-	d, err := crypto.Digest(data)
-	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
-	}
-	fmt.Println("Digest: ", d)
-	fmt.Println("")
-	//print line
-	printLine()
-
-	// Set Signers
-	signer := crypto.NewSigner(privatekey_pem, d)                   //default, using go's package
-	signerOpenSSL := crypto.NewSigner(privatekey_pem, d, "openssl") //openssl
-
-	// Sign
-	signature, err := signer.Sign()
-	if err != nil {
-		panic(err)
-	}
-	fmt.Println("Signature: ", signature)
-	fmt.Println("")
-
-	signatureOpenSSl, err := signerOpenSSL.Sign()
-	if err != nil {
-		panic(err)
-	}
-	fmt.Println("Signature OpenSSL: ", signatureOpenSSl)
-	fmt.Println("")
-
-	printLine()
-
-	// Test Verify correct public key
-	verifier := crypto.NewVerifier(publickey_pem, signature, data)
-	verifierOpenSSL := crypto.NewVerifier(publickey_pem, signatureOpenSSl, data, "openssl")
-
-	msg := "Verify correct public key: "
-
-	result := verifier.Verify()
-	printResult(msg, result)
-	fmt.Println("")
+	"flag"
 
-	result = verifierOpenSSL.Verify()
-	printResult("OpenSSL "+msg, result)
-	fmt.Println("")
-
-	printLine()
-
-	// Test Verify wrong public key
-	wrongPubKeyVerifier := crypto.NewVerifier("wrong public key", signature, data)
-	wrongPubKeyVerifierOpenSSL := crypto.NewVerifier("wrong public key", signatureOpenSSl, data, "openssl")
-
-	msg = "Result for wrong public key:"
-
-	result = wrongPubKeyVerifier.Verify()
-	printResult(msg, result)
-	fmt.Println("")
-
-	result = wrongPubKeyVerifierOpenSSL.Verify()
-	printResult("OpenSSL "+msg, result)
-	fmt.Println("")
+	"github.com/Xfers/cloudhsm-service-go/controllers"
+	_ "github.com/Xfers/cloudhsm-service-go/docs"
+	"github.com/gin-gonic/gin"
+	swaggerFiles "github.com/swaggo/files"
+	ginSwagger "github.com/swaggo/gin-swagger"
+)
 
-	printLine()
+// @title CloudHSM service API
+// @version 1.0
+// @description Standalone cloudhsm-service
 
-	// Test Verify wrong data
-	wrongDataVerifier := crypto.NewVerifier(publickey_pem, signature, "wrong data")
-	wrongDataVerifierOpenSSL := crypto.NewVerifier(publickey_pem, signatureOpenSSl, "wrong data", "openssl")
+// @host localhost:3000
+// @BasePath /
+// @schemes http
+func main() {
 
-	msg = "Result with wrong data:"
+	// Run test
+	// test()
 
-	result = wrongDataVerifier.Verify()
-	printResult(msg, result)
-	fmt.Println("")
+	//set flag for private key path
+	privateKeyPath := flag.String("private_key_path", "", "Path to private key")
+	// Set flag for public key path
+	publicKeyPath := flag.String("public_key_path", "", "Path to public key")
```

Comment:
I would suggest checking if key file path exists before server start

Reply:
great suggestions,. thanks.. I will work on them.
It is still a work in progress, I am still gathering requirements, and information. But this is why I put the PR ASAP and assign you as a review so I can get feedbacks earlier... thanks

Reply:
Since the parsed result instance are read-only and it only needs to be set up 1 time in application startup. So it dosen't even need to be thread-safety. A simple singleton structure instance should be enough.

Reply:
yeah, that makes sense. thanks, good catch... It is read only, only at the server start. So yeah, I will add a simple singleton, that parse the keys at the startup, and store them, to be used by different goroutines that are spawn at each request.